src/fabric.c

```diff
@@ -554,6 +554,13 @@ int DEFAULT_SYMVER_PRE(fi_getinfo)(uint32_t version, const char *node, const cha
 			continue;
 		}
 
+		if (!cur) {
```

Comment:
@shefty There's some core changes in here too. Does this seem like an appropriate way to handle this situation?